examples/ConfigurationStoreBackup/BackupAppConfigurationStore.cs

```diff
@@ -0,0 +1,184 @@
+                    JToken dataObject = JObject.Parse(decodedMessage).Property("data")?.Value;
+                    if (dataObject != null && dataObject["key"] != null)
+                    {
+                        string key = dataObject["key"].ToString();
+                        string label = dataObject["label"]?.ToString();
+                        updatedKeyLabels.Add(new KeyLabel(key, label));
+                    }
+                }
+                catch (FormatException)
+                {
+                    log.LogInformation($"Queue message in invalid Base64 format is ignored.\nMessage: {message.MessageText}");
+                }
+                catch (JsonReaderException)
+                {
+                    log.LogInformation($"Queue message in invalid JSON format is ignored.\nMessage: {decodedMessage}");
+                }
+            }
+            return updatedKeyLabels;
+        }
+
+        private static async Task BackupKeyValuesAsync(HashSet<KeyLabel> updatedKeyLabels,
+                                                       ConfigurationClient primaryAppConfigClient,
+                                                       ConfigurationClient secondaryAppConfigClient,
+                                                       ILogger log)
+        {
+            try
+            {
+                // Read all settings from primary store and update secondary store.
+                await foreach (ConfigurationSetting setting in primaryAppConfigClient.GetConfigurationSettingsAsync(new SettingSelector()))
+                {
+                    KeyLabel primaryStoreKeyLabel = new KeyLabel(setting.Key, setting.Label);
+                    if (updatedKeyLabels.Contains(primaryStoreKeyLabel))
+                    {
+                        // Current setting retrieved from primary store needs to be updated in secondary store.
+                        await secondaryAppConfigClient.SetConfigurationSettingAsync(setting);
+                        log.LogInformation($"Successfully updated key: {setting.Key} label: {setting.Label}");
+
+                        updatedKeyLabels.Remove(primaryStoreKeyLabel);
+                        if (updatedKeyLabels.Count == 0)
+                        {
+                            break;
+                        }
+                    }
+                }
+
+                // Delete key labels still present in updatedKeyLabels from secondary store.
+                foreach (var keyLabel in updatedKeyLabels)
+                {
+                    await secondaryAppConfigClient.DeleteConfigurationSettingAsync(keyLabel.Key, keyLabel.Label);
+                    log.LogInformation($"Successfully deleted key: {keyLabel.Key} label: {keyLabel.Label}");
+                }
+            }
+            catch (RequestFailedException)
+            {
+                log.LogError($"Request to Azure App Configuration failed.");
+                throw;
```

Comment:
Why we change this? I thought we didn't want to crash the Function for transient errors.